### PR TITLE
feat/add pre commit linter

### PR DESCRIPTION
**Pull request by 5h4z4mm**
**Description**

Setup pre-commit hook to lint source files.

- Fixes [#6](https://app.dework.xyz/smock/main-project-3127?taskId=b933bda9-7f3b-4d04-9e5e-8f22b5d7c56f)

**Test Evidence**
gitdiff

precommit

final

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -83,6 +89,7 @@
     "hardhat-preprocessor": "0.1.4",
     "husky": "6.0.0",
     "inquirer": "8.1.0",
+    "lint-staged": ">=10",
```
Comment (wei3erHase) [inline package.json:92]: precise versioning 🙏

**Review by wei3erHase — CHANGES_REQUESTED** [review]
yo @5h4z4mm , this looks nice!
please, verify that the unit test should pass, and if you can replicate the workflow trigger to activate linter check on pull requests would be nice, i just realize it doesn't run on this PR

File: package.json (hunk 1 above)
Comment (5h4z4mm) [reply package.json]: I've now tagged this dependency to version **12.5.0** which is the maximum allowed version for node engine **12.x**

**Comment by 5h4z4mm** [comment]
@wei3erHase workflow trigger is now updated to trigger linter check on pull request.

**Review by wei3erHase — APPROVED** [review]
works like a charm! 🪄 
thank you so much for contributing to smock!! 🚀 
there are a few more bounties for ya, wish you the best of lucks in this journey 🍀

**Pull request merged**